### ISSUE TITLE
Minecart Mount position Update issue.

### ISSUE THREAD

**Issue by tlworks** [Pryaxis/TShock#1243]
As shown in this image below:
https://i.gyazo.com/54e5d8ccbb96fe293b67cc27aed23f38.png

The Green zone is the Death Zone, the Blue zone is the zone next to the Death Zone & the Red zone is the zone above the Death Zone. Basically if someone drives a Minecart next to the Death Zone, nothing happens but if someone drives above the Death Zone with a Minecart, they eventually die because of their position not being updated properly.

It seems to be down to where the server thinks the player is. The position of the player due to the servers own updates puts the player inside the region, as though there's no code to keep the player on the track. As soon as the player moves, it thinks that the player is out of the region again, and when the player stops still, the server thinks the player is inside the region again, which is below the track.

We tried the UFO mount and it didn't update to be inside the region, which is the same as for using a hover board, hovering above the Death Zone. Additionally when adding blocks BELOW the Minecart track, the player never dies because it assumes the player 'falls' on solid ground, above the region.

The Death Zone plugin (RegionFlags) uses the TSPlayer.TileX and TSPlayer.TileY properties.

**Comment by Kojirremer** [comment]
I don't know if this issue is still there without the plugin, but I'm pretty sure its been stated that RegionFlags currently has a truckload of serious issues, ranging from region troubles to even causing the server time to go mad. Maybe it's time to find an alternative plugin, like RegionTrigger or SmartRegions.

**Comment by Patrikkk** [comment]
This kinda sounds plugin related. :|

**Comment by ProfessorXZ** [comment]
Perhaps, but may not be. @tlworks please test the possible fix I sent and reply back.

**Comment by nicatronTg** [comment]
@ProfessorXZ so what happened with this?

**Comment by nicatronTg** [comment]
_I am assigned to this issue for monitoring/oversight but am not actively working this issue._

**Comment by ProfessorXZ** [comment]
@nicatronTg I never got feedback from tlworks, thus I'll assume it is somehow plugin related. I am up for keeping this open however, will try to do some testing later on to confirm this.

**Comment by nicatronTg** [comment]
@ProfessorXZ roger.

**Comment by nicatronTg** [comment]
@ProfessorXZ ding dong what's the wring rong on this issue dong?

**Comment by DogooFalchion** [comment]
I am assuming that this issue is talking about when on a minecart/mount TShock doesn't know what the actual position of the client is?  Would need to put some logging in to the update/playerupdate to track the position and see if the server is doing some sort of bouncing, or if the plugin that is having issues is a problem.

**Comment by ProfessorXZ** [comment]
Confirmed issue; working on it.

**Comment by ProfessorXZ** [comment]
I've been pretty useless for the past few months so here: @DogooFalchion yeah the client is doing some weird bouncing. This only happens with the minecart mount IIRC. There is a method inside the Player class that checks whether the player is riding a minecart; if this is true the `fallThrough` flag is also set to true. This will then cause the player to "fall" until he hits solid ground. The position will then randomly change every few seconds, as if the player is jumping up. I'll attach a pic of what I have below.

![terraria_i_don t_know_that--_aaaaa _2016-10-30_23-04-52](https://cloud.githubusercontent.com/assets/7541476/21076330/d110434c-bf28-11e6-941d-1fe7efca93b4.png)

I've found no reliable way to get the actual position of the client yet, but the last time I actually did something regarding this issue was early November.
